### PR TITLE
planner: fix panic during optimizer trace

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31537

### What is changed and how it works?

fix panic during optimizer trace

### Check List

Tests 

- [x] Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: planner/core/find_best_task.go [hunk 1]
```diff
@@ -330,21 +330,21 @@ func (op *physicalOptimizeOp) appendChildToCandidate(candidateInfo *tracing.Phys
...
 	}
-	traceInfo.BestTask = &tracing.PhysicalPlanTrace{ID: pp.ID(), TP: pp.TP(), Info: pp.ExplainInfo(), Cost: pp.Cost()}
+	traceInfo.BestTask = &tracing.PhysicalPlanTrace{ID: pp.ID(), TP: pp.TP(), Cost: pp.Cost(), Info: pp.ExplainInfo()}
 }
 
 func (op *physicalOptimizeOp) appendCandidate(logicalPlan *baseLogicalPlan, physicalPlan PhysicalPlan, prop string) *tracing.PhysicalPlanTrace {
 	if op == nil || op.tracer == nil {
 		return nil
 	}
-	PhysicalPlanTrace := &tracing.PhysicalPlanTrace{TP: physicalPlan.TP(), ID: physicalPlan.ID(), Info: physicalPlan.ExplainInfo()}
```
Comment (Yisaer) [inline planner/core/find_best_task.go:347]: during `appendCandidate`, The info might be not initialized that would cause panic here

**Review by rebelice — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**